### PR TITLE
Adds a base class for controllers

### PR DESCRIPTION
**Pull request by thinkingmedia**
Noticed last time that the controllers to be reduced a little if they used a common base class.

### CODE REVIEW

**Comment by thinkingmedia** [comment]
So apparently the class identifier is not supported in PHP 5.4.

Isn't 5.4 below the required PHP version for Cake 3?

**Comment by ADmad** [comment]
> Isn't 5.4 below the required PHP version for Cake 3?

Cake 3.2 and below support 5.4 and hence Debugkit needs to work on PHP 5.4

**Comment by thinkingmedia** [comment]
done.

File: src/Controller/RequestsController.php [hunk 1]
```diff
@@ -34,12 +31,18 @@ class RequestsController extends Controller
...
+
+        parent::beforeFilter($event);
+    }
+
+    /**
+     * Do not load components for this controller.
+     *
+     * @return void
+     */
+    public function initialize()
```
Comment (markstory) [inline src/Controller/RequestsController.php]: Is the initialize state really 'common' if some of the controllers turn it off?

Comment (thinkingmedia) [inline src/Controller/RequestsController.php]: I thought it was, but that one controller has issues with the request handler. There are tests that say it's a JSON response but it has an HTML only view. Either I dig deep into the code to see why or revert the behavior for this one case.

**Comment by thinkingmedia** [comment]
closing as not important.

**Pull request closed without merging**